src/Tribe/Repositories/Traits/Post_Tickets.php

```diff
@@ -0,0 +1,292 @@
+<?php
+/**
+ * Post tickets trait that contains all of the ORM filters that can be used for any repository.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Tickets\Repositories\Traits
+ */
+
+namespace Tribe\Tickets\Repositories\Traits;
+
+use Tribe__Repository;
+use Tribe__Repository__Usage_Error;
+use Tribe__Repository__Void_Query_Exception;
+use Tribe__Utils__Array;
+
+/**
+ * Class Post_Tickets
+ *
+ * @since TBD
+ */
+trait Post_Tickets {
+
+	/**
+	 * A re-implementation of the base `filter_by_cost` method to filter events by related
+	 * ticket costs in place of their own cost meta.
+	 *
+	 * @since TBD
+	 *
+	 * @param float|array $value       The cost to use for the comparison; in the case of `BETWEEN`, `NOT BETWEEN`,
+	 *                                 `IN` and `NOT IN` operators this value should be an array.
+	 * @param string      $operator    Teh comparison operator to use for the comparison, one of `<`, `<=`, `>`, `>=`,
+	 *                                 `=`, `BETWEEN`, `NOT BETWEEN`, `IN`, `NOT IN`.
+	 * @param string      $symbol      The desired currency symbol or symbols; this symbol can be a currency ISO code,
+	 *                                 e.g. "USD" for U.S. dollars, or a currency symbol, e.g. "$".
+	 *                                 In the latter case results will include any event with the matching currency
+	 *                                 symbol, this might lead to ambiguous results.
+	 *
+	 * @throws Tribe__Repository__Usage_Error If the comparison operator is not supported of is using the `BETWEEN`,
+	 *                                        `NOT BETWEEN` operators without passing a two element array `$value`.
+	 */
+	public function filter_by_cost( $value, $operator = '=', $symbol = null ) {
+		$repo = $this;
+
+		// If the repo is decorated, use that.
+		if ( ! empty( $repo ) ) {
+			$repo = $this->decorated;
+		}
+
+		$operators = [
+			'<',
+			'<=',
+			'>',
+			'>=',
+			'=',
+			'!=',
+			'BETWEEN',
+			'NOT BETWEEN',
+			'IN',
+			'NOT IN',
+		];
+		if ( ! in_array( $operator, $operators, true ) ) {
+			throw Tribe__Repository__Usage_Error::because_this_comparison_operator_is_not_supported( $operator, 'filter_by_cost' );
+		}
+
+		if (
+			in_array( $operator, [ 'BETWEEN', 'NOT BETWEEN' ], true ) 
+			&& ! (
+				is_array( $value ) 
+				&& 2 === count( $value )
+			) 
+		) {
+			throw Tribe__Repository__Usage_Error::because_this_comparison_operator_requires_an_value_of_type( $operator, 'filter_by_cost', 'array' );
+		}
+
+		if ( in_array( $operator, [ 'IN', 'NOT IN' ], true ) ) {
+			$value = (array) $value;
+		}
+
+		$operator_name = Tribe__Utils__Array::get( Tribe__Repository::get_comparison_operators(), $operator, '' );
+		$prefix        = str_replace( '-', '_', 'by_cost_' . $operator_name );
+
```

Comment:
:no_entry_sign: This code pattern is often used to run a very dangerous shell programs on your server. The code in these files needs to be reviewed, and possibly cleaned.

<sub>hash: 0b2fb16e0a30fdd8d9426a4caaa7f0ec</sub>